docs/02.API-REFERENCE.md

```diff
@@ -3877,3 +3879,75 @@ jerry_parse_and_save_literals (const jerry_char_t *source_p,
 - [jerry_init](#jerry_init)
 - [jerry_cleanup](#jerry_cleanup)
 - [jerry_register_magic_strings](#jerry_register_magic_strings)
+
+
+## jerry_set_vm_exec_stop_callback
+
+**Summary**
+
+When JERRY_FEATURE_VM_EXEC_STOP is enabled a callback function can be
+specified by this function. This callback is periodically called when
+JerryScript executes an ECMASCript program.
+
+If the callback returns with undefined value the ECMAScript execution
+continues. Otherwise the result is converted to an error and thrown
```

Comment:
This would make the return value processing a bit more difficult since we would need to free the returned value, but would make the code more robust. If this is an acceptable overhead, I can change it.